src/backend/distributed/planner/shard_pruning.c

```diff
@@ -1563,3 +1563,24 @@ ExhaustivePruneOne(ShardInterval *curInterval,
 
 	return false;
 }
+
+
+/*
+ * ShardsForPartitionColumnValue gets the shards for a single partition column value.
+ */
+List *
+ShardsForPartitionColumnValue(Oid relationId, Node *partitionColumnValue)
+{
```

Comment:
Minor: Might be cheaper to call `FindShardInterval()` and avoids creating the `opExpr` etc. You can see `get_shard_id_for_distribution_column()` UDF for how to use it.
```C	
	DistTableCacheEntry *cacheEntry = DistributedTableCacheEntry(relationId);
    ... calculate distributionValueDatum ...
	shardInterval = FindShardInterval(distributionValueDatum, cacheEntry);
```